### PR TITLE
[#3688] Remove `productaanvraagType` for v2 options

### PR DESCRIPTION
**Pull request by Viicos**
Part of #3688

### CODE REVIEW

File: src/openforms/js/components/admin/form_design/registrations/objectsapi/ObjectsApiOptionsFormFields.js [hunk 1]
```diff
@@ -47,6 +47,7 @@ const ObjectsApiOptionsFormFields = ({index, name, schema, formData, onChange})
         } else {
           delete draft.variablesMapping;
         }
+        delete draft.productaanvraagType;
```
Comment (SilviaAmAm) [inline src/openforms/js/components/admin/form_design/registrations/objectsapi/ObjectsApiOptionsFormFields.js:50]: Maybe nice to add a check in the story that this field is not present?

**Review by SilviaAmAm — APPROVED** [review]

**Review by vaszig — COMMENTED** [review]
Looks good. I think we missed only the removal of `productaanvraagType` in the definition (`V2ConfigFields.propTypes`)

File: src/openforms/js/components/admin/form_design/registrations/objectsapi/ObjectsApiOptionsFormFields.js (hunk 1 above)
Comment (sergei-maertens) [reply src/openforms/js/components/admin/form_design/registrations/objectsapi/ObjectsApiOptionsFormFields.js:50]: Those kind of checks are a bit fragile, you can easily satisfy them by changing a translation to something slightly different. I trust the review process more on this :)

**Review by sergei-maertens — APPROVED** [review]
Processed the feedback (proptype removal)

**Pull request merged**